master/buildbot/worker/openstack.py

```diff
@@ -330,6 +343,25 @@ def stop_instance(self, fast=False):
         self._stop_instance(instance, fast)
         return None
 
+    def _stop_all_instances(self):
```

Comment:
I think you can probably use this implementation for _stop_instance as well

Reply:
_stop_instance looks up things by instance.id, this one looks them by name.

Reply:
right, but we don't really need both. If for some reason we loose track of ID, we can still stop the instance if we don't change the worker name

Reply:
@tardyp  is this now what you wanted?